src/index.js

```diff
@@ -197,45 +197,70 @@ async function getDocuments() {
+      tmpDocs = Array.from(
+        $year('.documents')
+          .find('ul[class="list-unstyled documents"] > li')
+          .map((idx, el) => {
+            let label = $year(el)
+              .find('div.hidden-xs.texte > span')
               .text()
               .trim()
-          }
-          if (label.match(/Décla\s/g)) {
-            log('debug', 'getting in décla matching condition')
-            label = label.replace('Décla', 'Déclaration')
-          }
-          const idEnsua = $year(el).find('input').attr('value')
-          let filename = `${year}-${label}.pdf`
-          // Replace / and : found in some labels
-          // 1) in date (01/01/2018 -> 01-01-2018)
-          filename = filename.replace(/\//g, '-')
-          // 2) in complementrary form
-          filename = filename.replace(' : ', ' - ') // eslint-disable-line
-          filename = filename.replace(' : ', ' - ')
-          // 3) replace time (19:26 -> 19h26)
-          filename = filename.replace(':', 'h')
-          return {
-            year,
-            label,
-            idEnsua,
-            filename,
-            fileurl:
-              `https://cfspart.impots.gouv.fr/enp/Affichage_Document_PDF` +
-              `?idEnsua=${idEnsua}`
-          }
-        })
-    )
-    log('info', `${tmpDocs.length} docs found for year ${year}`)
+            if (label.length === 0) {
+              label = $year(el)
+                .find('div[class="visible-xs col-xs-5 texte_docslies"] > span')
+                .text()
+                .trim()
+            }
+            if (label.match(/Décla\s/g)) {
+              log('debug', 'getting in décla matching condition')
+              label = label.replace('Décla', 'Déclaration')
+            }
+            const idEnsua = $year(el).find('input').attr('value')
+            let filename = `${year}-${label}.pdf`
+            // Replace / and : found in some labels
+            // 1) in date (01/01/2018 -> 01-01-2018)
+            filename = filename.replace(/\//g, '-')
+            // 2) in complementrary form
+            filename = filename.replace(' : ', ' - ') // eslint-disable-line
+            filename = filename.replace(' : ', ' - ')
+            // 3) replace time (19:26 -> 19h26)
+            filename = filename.replace(':', 'h')
+            return {
+              year,
+              label,
+              idEnsua,
+              filename,
+              fileurl:
+                `https://cfspart.impots.gouv.fr/enp/Affichage_Document_PDF` +
+                `?idEnsua=${idEnsua}`
+            }
+          })
+      )
+      if (!baseLength) {
+        log('debug', 'First baseLength definition')
+        baseLength = tmpDocs.length
+      } else {
+        log('debug', 'testLength definition')
+        testLength = tmpDocs.length
+      }
+      if (testLength > baseLength) {
+        log('debug', 'testLength is greater, becoming baseLength')
+        baseLength = testLength
+      }
+      loop++
```

Comment:
Principaly to avoid this 3rd loop if not necessary, if the same number of bills is found on second call, then it's most likely that the website has loaded everything. For me (my account, on local) it often happened that the first loop was not returning every files for a year, but on the other hand I've never done the 3rd loop, it is just a precaution.